### PR TITLE
GDPR compliance: data section of wizard

### PR DESCRIPTION
**Pull request by sampling**
This is the first part of changes in the scope of the ticket:
https://trello.com/c/kkuQxS8T/224-gdpr-compliance-for-our-data-step

(second part will be to remove the "How will data be used?" field and the third one will be to involve shared_with value into information sheet, as this moment it is completely ignored)

@joelcarr @rgarner please take a look

### CODE REVIEW

**Comment by rgarner** [comment]
Hi @sampling – I've renamed the PR, because "form builder" means something [different](https://github.com/barnardos/consent-form-builder-rails/blob/master/app/helpers/barnardos/action_view/form_builder.rb#L3) internally to the project, and if this PR is part of wider GDPR work it will hopefully be helpful to see that at the start of the title (also we are trying to avoid the word "changes" but I'm not sure if we've been explicit about that in the devs-manual yet).

**Comment by rgarner** [comment]
I like that you've gone straight for the `I18n` work – it seems very sensible to start to do this a bit at a time rather than the [big bang](https://trello.com/c/VtkrQFmq/178-move-every-piece-of-copy-to-enyml) effort we've talked about before.

File: app/models/shared_with.rb [hunk 1]
```diff
@@ -1,8 +1,9 @@
 class SharedWith
   NAME_VALUES = HashWithIndifferentAccess.new(
-    team:     'Just the team',
-    internal: 'Other teams internally',
-    external:  'Other teams externally'
+    anonymised: I18n.t('helpers.label.storing_information.shared_with.anonymised'),
+    team:       I18n.t('helpers.label.storing_information.shared_with.team'),
+    internal:   I18n.t('helpers.label.storing_information.shared_with.internal'),
+    external:   I18n.t('helpers.label.storing_information.shared_with.xternal')
```
Comment (rgarner) [inline app/models/shared_with.rb]: Slight typo here: `xternal`

Comment (rgarner) [reply app/models/shared_with.rb]: (practical upshot: you'll see a "translation missing" message in the HTML source of the output and ActiveSupport will try to infer something reasonable)

File: spec/helpers/barnardos/action_view/form_builder_spec.rb [hunk 2]
```diff
@@ -67,9 +67,10 @@ class TestHelperCarrier < ActionView::Base
 
     let(:collection) do
       {
-        team:     'Just the team',
-        internal: 'Other teams internally',
-        external: 'Other teams externally'
+        anonymised: 'The first label',
+        team:       'The second label',
+        internal:   'The third label',
+        external:   'The forth label'
```
Comment (rgarner) [inline spec/helpers/barnardos/action_view/form_builder_spec.rb]: Slight typo, should read 'the fourth label'. Affects nothing. Except for me writing this comment.

File: app/views/research_sessions/data.html.erb [hunk 3]
```diff
@@ -1,18 +1,18 @@
 <%= render "research_sessions/progress" %>
 <div class="main-content">
   <%= render 'error_summary' %>
-  <h1 class="subtitle-large">Data</h1>
+  <h1 class="subtitle-large">Storing information</h1>
```
Comment (rgarner) [inline app/views/research_sessions/data.html.erb]: If we're going to say "storing" here, and "storing" in the progress bar, we should probably rename this to `storing.html.erb` and also update the wizard [`Steps`](https://github.com/barnardos/consent-form-builder-rails/blob/master/app/models/research_session/steps.rb#L14) to keep our domain language consistent.

**Comment by rgarner** [comment]
Overall I'd say there's a ton of really useful stuff in here, for which, thank you @sampling and @Jesskyri. We need a bit of a rebase to tidy up the grouping of the commits and their [messages](https://github.com/barnardos/devs-manual/blob/master/styleguides/git.md#commit-messages) but this is all darn useful.

File: app/models/shared_with.rb [hunk 4]
```diff
@@ -1,8 +1,9 @@
 class SharedWith
   NAME_VALUES = HashWithIndifferentAccess.new(
-    team:     'Just the team',
-    internal: 'Other teams internally',
-    external:  'Other teams externally'
+    anonymised: 'All identifiable information will be anonymised as we process it',
+    team:       'Identifiable information will only be shared with the research team at Barnardo\'s',
```
Comment (rgarner) [inline app/models/shared_with.rb]: You can avoid the need to escape the apostrophe by using double quotes on this string and others like it. If you feel like it.

File: app/views/research_sessions/data.html.erb (hunk 3 above)
Comment (rgarner) [reply app/views/research_sessions/data.html.erb]: It looks like if we update [`Steps`](https://github.com/barnardos/consent-form-builder-rails/blob/master/app/models/research_session/steps.rb#L14) as suggested above, that will also change the text in the [progress bar](https://github.com/barnardos/consent-form-builder-rails/blob/master/app/views/research_sessions/_progress.html.erb#L26) – we should probably keep this stuff in line in this PR 😄

Comment (sampling) [reply app/views/research_sessions/data.html.erb]: The progress bar part is covered by different card where all the navigation changes are included

Comment (rgarner) [reply app/views/research_sessions/data.html.erb]: That's fine but the step name should probably still remain consistent with the copy in this PR. That would mean the progress bar changing at the same time as a side-effect.

**Comment by rgarner** [comment]
I understand now why the commits are sparse, if you've done this without a local environment – that's truly impressive. I'm going to call you the next time I need my hall wallpapering when I only have access through my letterbox.

**Comment by rgarner** [comment]
This PR is superseded by #169 for the acceptance criteria detailed therein. This PR still has changes that are not related to [this card](https://trello.com/c/kkuQxS8T/224-gdpr-compliance-for-our-data-now-storing-step), and should be analysed for outstanding changes. Those changes should be extracted to separate cards.

**Pull request closed without merging**